source/common/stats/tag_extractor_impl.cc

```diff
@@ -169,5 +180,89 @@ bool TagExtractorRe2Impl::extractTag(absl::string_view stat_name, std::vector<Ta
   return false;
 }
 
+TagExtractorTokensImpl::TagExtractorTokensImpl(absl::string_view name, absl::string_view tokens,
+                                               absl::string_view substr)
+    : TagExtractorImplBase(name, tokens, substr), tokens_(absl::StrSplit(tokens, '.')),
+      match_index_(findMatchIndex(tokens_)) {
+  prefix_ = tokens_[0];
```

Comment:
Is it ok to have `$` as a prefix if it's in the zeroth position?

Reply:
good catch: this was a bug. Also `*` and `**`. Added tests.